### PR TITLE
fix: flaky test `Settings Redirects to ENS domains when user inputs ENS into address bar`

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
This PR fixes some flakiness of the test `Settings Redirects to ENS domains when user inputs ENS into address bar`. After looking into the code I've identified a source of flakiness, as well as a problem that prevents to upload the artifacts (screenshot) if the test fail.

- The test was not using any mock: this is a source of flakiness, and it's now fixed by adding the necessary mocks to the ENS contracts and Infura calls (we need to set Mainnet, since this feature is only available on Mainnet)
- The test was not using fixtures: this per se is not a problem, but the issue here is that we are not triggering the `verboseReportOnFailure(title,error)` on error, making it harder to debug, as no artifacts are uploaded for this test, if it fails. Now we are going to be able to see the screenshot if the test ever fails again

Circle ci job: https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/83569/workflows/0883aa51-cd94-4066-9f96-77d1375e5e9b/jobs/2994904/tests

Failure log: `TimeoutError: Wait timed out after 213ms`

Extra note: @davidmurdoch is also working on mitigating the rest of the flakiness by waiting for the metamask to be fully initialized before proceeding with the test. This work will be done in a separate PR.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24898?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

See no artifact file was uploaded:

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/83569/workflows/0883aa51-cd94-4066-9f96-77d1375e5e9b/jobs/2994904/artifacts

Screenshot from 2024-05-30 13-43-01

Screenshot from 2024-05-30 13-43-04

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/settings/ipfs-ens-resolution.spec.js [hunk 1]
```diff
@@ -10,25 +155,36 @@ describe('Settings', function () {
...
-      // since all we care about is getting to the correct URL
-    }
-
-    // Ensure that the redirect to ENS Domains has happened
-    await driver.wait(async () => {
-      const currentUrl = await driver.getCurrentUrl();
-      return currentUrl === ENS_DESTINATION_URL;
-    }, tinyDelayMs);
-
-    await driver.quit();
```
Comment (seaona) [inline test/e2e/tests/settings/ipfs-ens-resolution.spec.js:31]: no need to start or stop the driver manually, as now it's done within the fixtures

File: test/e2e/tests/settings/ipfs-ens-resolution.spec.js [hunk 2]
```diff
@@ -10,25 +155,36 @@ describe('Settings', function () {
   it('Redirects to ENS domains when user inputs ENS into address bar', async function () {
     // Using proxy port that doesn't resolve so that the browser can error out properly
     // on the ".eth" hostname.  The proxy does too much interference with 8000.
-    const { driver } = await buildWebDriver({ proxyUrl: '127.0.0.1:8001' });
```
Comment (seaona) [inline test/e2e/tests/settings/ipfs-ens-resolution.spec.js:13]: to preserve the proxy number, we now pass this as an option in fixtures, and forward it to the chrome/firefox drivers as an option

**Review by DDDDDanica — DISMISSED** [review]

**Comment by hjetpoluru** [comment]
I am getting this error when I ran locally against firefox browser <?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="9.584" tests="2" failures="2">
  <testsuite name="Root Suite" timestamp="2024-05-30T20:21:30" tests="0" time="0.000" failures="0">
  </testsuite>
  <testsuite name="Settings" timestamp="2024-05-30T20:21:30" tests="2" file="/Users/harikajetpoluru/repo/metamask-extension/test/e2e/tests/settings/ipfs-ens-resolution.spec.js" time="9.583" failures="2">
    <testcase name="Settings Redirects to ENS domains when user inputs ENS into address bar" time="9.179" classname="Redirects to ENS domains when user inputs ENS into address bar">
      <failure message="Wait timed out after 212ms" type="TimeoutError"><![CDATA[TimeoutError: Wait timed out after 212ms
    at /Users/harikajetpoluru/repo/metamask-extension/node_modules/selenium-webdriver/lib/webdriver.js:929:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)]]></failure>
    </testcase>
    <testcase name="Settings Does not fetch ENS data for ENS Domain when ENS and IPFS switched off" time="0.342" classname="Does not fetch ENS data for ENS Domain when ENS and IPFS switched off">
      <failure message="listen EADDRINUSE: address already in use 127.0.0.1:8545." type="Error"><![CDATA[Error: listen EADDRINUSE: address already in use 127.0.0.1:8545.
    at /Users/harikajetpoluru/repo/metamask-extension/node_modules/ganache/dist/node/webpack:/Ganache/core/lib/src/server.js:193:33
    at async Promise.allSettled (index 1)
    at Ganache.start (test/e2e/seeder/ganache.ts:19:9)
    at withFixtures (test/e2e/helpers.js:29:55)
    at Context.<anonymous> (test/e2e/tests/settings/ipfs-ens-resolution.spec.js:174:9)]]></failure>
    </testcase>
  </testsuite>
</testsuites>

File: test/e2e/webdriver/chrome.js [hunk 3]
```diff
@@ -3,21 +3,39 @@ const chrome = require('selenium-webdriver/chrome');
...
+ */
+const { SELENIUM_HTTPS_PROXY } = process.env;
+
+/**
+ * Determine the appropriate proxy server value to use
+ *
+ * @param {string|number} [proxyPort] - The proxy port to use
+ * @returns {string} The proxy server address
+ */
+function getProxyServer(proxyPort) {
```
Comment (legobeat) [inline test/e2e/webdriver/chrome.js:25]: If the function is there to get the right value, the constants are just footguns to expose? So can be moved into the inner scope.

```suggestion
/**
 * Determine the appropriate proxy server value to use
 *
 * @param {string|number} [proxyPort] - The proxy port to use
 * @returns {string} The proxy server address
 */
function getProxyServer(proxyPort) {
  const DEFAULT_PROXY_HOST = '127.0.0.1:8000';
  const { SELENIUM_HTTPS_PROXY } = process.env;
```

File: test/e2e/webdriver/firefox.js [hunk 4]
```diff
@@ -19,11 +19,25 @@ const { retry } = require('../../../development/lib/retry');
...
 /**
- * Proxy host to use for HTTPS requests
+ * Default proxy host to use for HTTPS requests
+ *
+ * @type {string}
  */
-const HTTPS_PROXY_HOST = new URL(
-  process.env.SELENIUM_HTTPS_PROXY || 'http://127.0.0.1:8000',
-);
+const DEFAULT_PROXY_HOST = 'http://127.0.0.1:8000';
```
Comment (legobeat) [inline test/e2e/webdriver/firefox.js:26]: se chrome comment about scope

Comment (seaona) [reply test/e2e/webdriver/firefox.js]: thank you that's a great suggestion :heart_eyes: !! I've implemented the changes

**Comment by seaona** [comment]
> @seaona, I am getting this error when I ran locally against firefox and its not consistent.
hey @hjetpoluru , thank you for reporting this :bowing_woman:  I'll trigger a couple of more ci runs to double-check it does not fail and report back here :pray:

**Review by legobeat — DISMISSED** [review]

**Comment by hjetpoluru** [comment]
@seaona , Good part is that - the issue is partially fixed - we are now able to obtain the failure details in the artifacts. However, the downside is that it's still flaky.https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/84316/workflows/2cea2ca2-c1d0-4e82-96dc-a51859d177dc/jobs/3033481/artifacts

**Comment by seaona** [comment]
yeah @hjetpoluru I move this to draft to investigate further the flakiness observed in ci, now that we have screenshots available with the current fix :+1: thank you!!

**Review by HowardBraham — DISMISSED** [review]

**Comment by seaona** [comment]
> Needs a little more work


The redirect flakiness has now been mitigated using @HowardBraham 's suggestion with the use of a greater max value.
The PR is now ready for review

**Review by chloeYue — APPROVED** [review]

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**